src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -1456,15 +1481,12 @@ int64_t SingleStreamDecoder::getPts(int64_t frameIndex) {
 // STREAM AND METADATA APIS
 // --------------------------------------------------------------------------
 
-int64_t SingleStreamDecoder::getNumFrames(
+std::optional<int64_t> SingleStreamDecoder::getNumFrames(
     const StreamMetadata& streamMetadata) {
   switch (seekMode_) {
     case SeekMode::exact:
       return streamMetadata.numFramesFromScan.value();
     case SeekMode::approximate: {
-      TORCH_CHECK(
-          streamMetadata.numFrames.has_value(),
-          "Cannot use approximate mode since we couldn't find the number of frames from the metadata.");
       return streamMetadata.numFrames.value();
```

Comment:
I think we mean to return `streamMetadata.numFrames` instead of `streamMetadata.numFrames.value()`?

Reply:
Maybe I'm missing something but I'm surprised this compiles. This function seems to always return `int64_t`, never a `std::optional<int64_t>` ?

Reply:
Ah, it reminds me of this great random number generator:

```py
def get_random_int():
    return 4
```